e107_plugins/forum/shortcodes/batch/viewforum_shortcodes.php

```diff
@@ -347,9 +347,14 @@ function sc_perms()
 					$permDisplay['edit'] = LAN_FORUM_0048;
 				}
 			--*/
+			if(!empty($parm['type']))
+			{
+			  return 'alert alert-'.($forum->checkPerm($forumId, 'post')?'info':'danger');
+			}
 
-			return implode("<span class='forum-perms-separator'><!-- --></span>", $permDisplay);
-
+			global $FORUM_VIEWFORUM_TEMPLATE;
```

Comment:
Using global is very PHP5. Please load the template. It can be done in the construct()

Reply:
It's the way the forum code is written. The template is already loaded in the main forum_view.php code, and for me it was a overload to load the template twice...
There's another way to do it. Pass the template via sc var, but that would require rewrite the main php file code....
Anyway all the forum code needs a major rewrite, it's poorly programmed...

If i'm allowed, i can do it...

Reply:
Yeah, for quite some time I've been considering a rewrite from scratch of the forum plugin but while still keeping the same database structure.